Спринт 10 ТЗ

ntwzprac opened the pull request:
Снова забыл поменять аккаунт перед commit'ом

notweuz committed:
Спринт 10 ТЗ

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,63 @@
+    public Film createFilm(@Valid @RequestBody Film film) {
+        film.setId(getNextId());
+
+        if (film.getReleaseDate().isBefore(LocalDate.of(1895, 12, 28))) {
```
вынеси во внутренний метод класса и используй как при создании, так и обновлении

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,63 @@
+
+    @PostMapping
+    public Film createFilm(@Valid @RequestBody Film film) {
+        film.setId(getNextId());
```
присвоение id после валидации

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,62 @@
+    public User createUser(@Valid @RequestBody User user) {
+        user.setId(getNextId());
+
+        if (user.getName() == null || user.getName().isEmpty()) {
```
и это тоже во внутренний метод

ntwzprac committed:
Исправления ТЗ 10 (1)

Затронуты:
- FilmController.java
- UserController.java

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/controller/UserController.java:
```diff
@@ -0,0 +1,66 @@
+        return ++currentMaxId;
+    }
+
+    private void validateAndSetName(User user) {
```
при обновлении тоже нужно прогнать пользователя через этот метод

ntwzprac committed:
Исправления ТЗ 10 (2)

Затронуты:
- UserController.java

ntwzprac committed:
Исправлен тест UserControllerTest.java

- Ошибка до исправления:
UserControllerTest.updateUser_shouldUpdateExistingUser:98 expected: <Updated User> but was: <Test User>
